misc/add-lxc-iptag.sh

```diff
@@ -0,0 +1,351 @@
+        return 1
+    else
+        last_net_interface="${current_net_interface}"
+        return 0
+    fi
+}
+
+# =============== MAIN =============== #
+
+update_lxc_iptags() {
+    vmid_list=$(pct list 2>/dev/null | grep -v VMID | awk '{print $1}')
+    for vmid in ${vmid_list}; do
+        last_tagged_ips=()
+        current_valid_ips=()
+        next_tags=()
+
+        # Parse current tags
+        mapfile -t current_tags < <(pct config "${vmid}" | grep tags | awk '{print $2}' | sed 's/;/\n/g')
+        for current_tag in "${current_tags[@]}"; do
+            if is_valid_ipv4 "${current_tag}"; then
+                last_tagged_ips+=("${current_tag}")
+                continue
+            fi
+            next_tags+=("${current_tag}")
+        done
+
+        # Get current IPs
+        current_ips_full=$(lxc-info -n "${vmid}" -i | awk '{print $2}')
+        for ip in ${current_ips_full}; do
+            if is_valid_ipv4 "${ip}" && ip_in_cidrs "${ip}" "${CIDR_LIST[*]}"; then
+                current_valid_ips+=("${ip}")
+                next_tags+=("${ip}")
+            fi
+        done
+
+        # Skip if no ip change
+        if [[ "$(echo "${last_tagged_ips[@]}" | tr ' ' '\n' | sort -u)" == "$(echo "${current_valid_ips[@]}" | tr ' ' '\n' | sort -u)" ]]; then
+            echo "Skipping ${vmid} cause ip no changes"
+            continue
+        fi
+
+        # Set tags
+        echo "Setting ${vmid} tags from ${current_tags[*]} to ${next_tags[*]}"
+        pct set "${vmid}" -tags "$(IFS=';'; echo "${next_tags[*]}")"
+    done
+}
+
+check() {
+    current_time=$(date +%s)
+
+    time_since_last_lxc_status_check=$((current_time - last_lxc_status_check_time))
+    if [[ "${LXC_STATUS_CHECK_INTERVAL}" -gt 0 ]] \
+        && [[ "${time_since_last_lxc_status_check}" -ge "${STATUS_CHECK_INTERVAL}" ]]; then
+        echo "Checking lxc status..."
+        last_lxc_status_check_time=${current_time}
+        if lxc_status_changed; then
+            update_lxc_iptags
+            last_update_time=${current_time}
+            return
+        fi
+    fi
+
+    time_since_last_fw_net_interface_check=$((current_time - last_fw_net_interface_check_time))
+    if [[ "${FW_NET_INTERFACE_CHECK_INTERVAL}" -gt 0 ]] \
+        && [[ "${time_since_last_fw_net_interface_check}" -ge "${FW_NET_INTERFACE_CHECK_INTERVAL}" ]]; then
+        echo "Checking fw net interface..."
+        last_fw_net_interface_check_time=${current_time}
+        if fw_net_interface_changed; then
+            update_lxc_iptags
+            last_update_time=${current_time}
+            return
+        fi
+    fi
+
+    time_since_last_update=$((current_time - last_update_time))
+    if [ ${time_since_last_update} -ge ${FORCE_UPDATE_INTERVAL} ]; then
+        echo "Force updating lxc iptags..."
+        update_lxc_iptags
+        last_update_time=${current_time}
+        return
+    fi
+}
+
+# main: Set the IP tags for all LXC containers
+main() {
+    while true; do
```

Comment:
Hi, have you considered using `systemctl timers`/`cron` instead of a while loop?

Reply:
To be honest, I would prefer using a resident service rather than scheduled tasks. The main reasons are:

1. There are multiple INTERVAL configurations and some states; using scheduling might complicate things.
2. Scheduled tasks to me lean more towards `doing something at a specific time (timer)` rather than `doing something every certain period of time (ticker)`.
3. A resident task allows me to know it is still running, whereas a scheduled task relies on external scheduling, and I can only know it's alive when it runs next.
4. I don't want to rely too much on external components; for users, they could even use `nohup ./lxc-iptag &`.

Can you tell me, what considerations is your proposal based on? If you indeed only want it to run once, perhaps I can provide an option for "once," then you could use cron or any other scheduler to run it. Do you think this would be a better solution? 

However, I believe that for most users, the default of using a resident task should suffice.

> Of course, I'd prefer if it were an event-triggered task rather than querying status periodically, but unfortunately, I haven't found an event source yet.

Reply:
> Can you tell me, what considerations is your proposal based on?

using `systemctl` it would be easy to check when the next run will be perfomed, it would also be possible to easily disable this job or run it manually. Also it would out of the box keep logs in the system journal.

> Of course, I'd prefer if it were an event-triggered task rather than querying status periodically, but unfortunately, I haven't found an event source yet.

This is much better idea! What about using `ip monitor` inside a systemd service?

Reply:
Thank you for your suggestion, I will try (but sorry, maybe not now).

Welcome to submit a PR.